test/shared-schemas.test.js

```diff
@@ -441,6 +441,47 @@ test('Use the same schema id in diferent places', t => {
   })
 })
 
+test('Use shared schema and $ref with $id', t => {
+  t.plan(1)
+  const fastify = Fastify()
+
+  fastify.addSchema({
+    $id: 'test',
+    type: 'object',
+    properties: {
+      id: { type: 'number' }
+    }
+  })
+
+  const body = {
+    $schema: 'http://json-schema.org/draft-07/schema#',
+    $id: 'http://foo/user',
+    type: 'object',
+    definitions: {
+      address: {
+        $id: '#address',
+        type: 'object',
+        properties: {
+          city: { 'type': 'string' }
+        }
+      }
+    },
+    properties: {
+      test: 'test#',
+      address: { $ref: '#address' }
+    }
+  }
+
+  fastify.route({
+    method: 'POST',
+    url: '/',
+    schema: { body },
```

Comment:
can you add a test for the 200 status code, and see if it works?

```js
      response: {
        200: 'test#'
      }
```

Reply:
Good test.. if I put `$ref = #address` in the reponse the serialization breaks 😃

Keep investigating 🔍

> RangeError: Maximum call stack size exceeded
    at String.replace (<anonymous>)
    at nested (C:\Users\behem\workspace\fastify\node_modules\fast-json-stringify\index.js:801:40)
    at Object.keys.forEach (C:\Users\behem\workspace\fastify\node_modules\fast-json-stringify\index.js:501:20)

Reply:
I think that’s a bug, and I think we need to resolve it. It might be in fast-json-stringify.